### PR TITLE
Implement install library method for installing multi binary recipes

### PR DESCRIPTION
**Pull request by veren-damyanov**
Implement install library method for installing multi binary recipes
Rework recipes to use it

### CODE REVIEW

**Comment by veren-damyanov** [comment]
Recipes that have been reworked to use the new install library:
- golang-4dev
- nodejs-4dev
- pycharm-community

File: src/lib/install.sh [hunk 1]
```diff
@@ -5,8 +5,9 @@ import assert
 
 _install_name_='install.sh'
 
-#: $1 - path to source, $2 - name, $3 - version, $4 (optional) - relative path to binary (w/o the first dir)
 install_single() {
+    #: Used to install single-binary recipes
+    #: $1 - path to source, $2 - name, $3 - version, $4 (optional) - relative path to binary (w/o the first dir)
     echo "install_single($1, $2, $3, $4) called!"
```
Comment (ya55en) [inline src/lib/install.sh:11]: (Not part of this change but still necessary:) Please rewrite this and similar `echo`-based log statements using the logging module functions (like `_debug`, `_info`, etc.)

File: src/lib/install.sh [hunk 2]
```diff
@@ -21,6 +22,63 @@ install_single() {
...
+    #:    ($1) tarball_path - path to the source tarball
+    #:    ($2) recipe_name - the name of the recipe currently being installed
+    #:    ($3) version - the version of the recipe currently being installed
+    local tarball_path="$1"; local recipe_name="$2"; local version="$3"
+
+    #: create proper .local/opt/name/version directory structure and extract the tarball
+    local recipe_dir="$_LOCAL/opt/$recipe_name"
+    if [ -e "$recipe_dir/$version" ]; then
+        _warn "Current version seems to have been installed in $recipe_dir"
+        # exit here?
```
Comment (ya55en) [inline src/lib/install.sh:36]: I would say **yes**, we should exit here with a non-zero code (e.g. 9 for now to avoid collisions with lower values which may mean something else). Also, maybe adding something like "please remove it and try again".

**Review by ya55en — APPROVED** [review]
Looks good! There are couple non-critical suggestions that I would be happy to see reflected before merging.
